src/lib.rs

```diff
@@ -1207,20 +1215,22 @@ impl<T: Counter> Histogram<T> {
     ///
     /// Two values are considered "equivalent" if `self.equivalent` would return true.
     ///
-    /// Returns an error if index calculations yield an inexpressible `usize`.
+    /// If the value is larger than the maximum representable value, it will be clamped to the
+    /// max representable value.
     ///
-    /// Panics if the value is out of bounds.
-    pub fn percentile_below(&self, value: u64) -> Result<f64, ()> {
+    /// If the total count of the histogram has reached `u64::max_value()`, this will return
+    /// inaccurate results.
+    pub fn percentile_below(&self, value: u64) -> f64 {
         if self.total_count == 0 {
-            return Ok(100.0);
+            return 100.0;
         }
 
-        let target_index = cmp::min(self.index_for(value).ok_or(())?, self.last());
+        let target_index = self.index_for_or_last(value);
         // TODO panic on bad index
-        let total_to_current_index = (0..(target_index + 1))
+        let total_to_current_index = (0..target_index.checked_add(1).expect("usize overflow"))
```

Comment:
Ugh, this is unfortunate. What we really want is [`RangeInclusive`](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html), but it's currently nightly-only. Perhaps leave a TODO?

Reply:
Definitely. I've wanted this more than a few times. :(